tests/lib/rules/no-plusplus.js

```diff
@@ -26,14 +26,31 @@ ruleTester.run("no-plusplus", rule, {
         { code: "var foo = 0; foo=+1;", options: [{allowForLoopAfterthoughts: true}] },
         { code: "for (i = 0; i < l; i++) { console.log(i); }", options: [{allowForLoopAfterthoughts: true}] }
     ],
-
     invalid: [
-        { code: "var foo = 0; foo++;", errors: [{ message: "Unary operator '++' used.", type: "UpdateExpression"}] },
-        { code: "var foo = 0; foo--;", errors: [{ message: "Unary operator '--' used.", type: "UpdateExpression"}] },
-        { code: "for (i = 0; i < l; i++) { console.log(i); }", errors: [{ message: "Unary operator '++' used.", type: "UpdateExpression"}] },
+        {
+            code: "function foo() { return i++; }",
+            errors: [{ message: "Unary operator '++' used.", type: "UpdateExpression"}]
+        },
+        { code: "var foo = 0; foo++;", output: "var foo = 0; foo += 1;", errors: [{ message: "Unary operator '++' used.", type: "UpdateExpression"}] },
+        { code: "var foo = 0; foo--;", output: "var foo = 0; foo -= 1;", errors: [{ message: "Unary operator '--' used.", type: "UpdateExpression"}] },
+        {
+            code: "for (i = 0; i < l; i++) { console.log(i); }",
+            output: "for (i = 0; i < l; i += 1) { console.log(i); }",
+            errors: [{ message: "Unary operator '++' used.", type: "UpdateExpression"}]
+        },
 
         // With "allowForLoopAfterthoughts" allowed
-        { code: "var foo = 0; foo++;", options: [{allowForLoopAfterthoughts: true}], errors: [{ message: "Unary operator '++' used.", type: "UpdateExpression"}] },
-        { code: "for (i = 0; i < l; i++) { v++; }", options: [{allowForLoopAfterthoughts: true}], errors: [{ message: "Unary operator '++' used.", type: "UpdateExpression"}] }
+        {
+            code: "var foo = 0; foo++;",
+            output: "var foo = 0; foo += 1;",
+            options: [{allowForLoopAfterthoughts: true}],
+            errors: [{ message: "Unary operator '++' used.", type: "UpdateExpression"}]
+        },
+        {
+            code: "for (i = 0; i < l; i++) { v++; }",
+            output: "for (i = 0; i < l; i++) { v += 1; }",
+            options: [{allowForLoopAfterthoughts: true}],
+            errors: [{ message: "Unary operator '++' used.", type: "UpdateExpression"}]
+        }
```

Comment:
Can you add tests for cases like `++i` and `--i` as well? I think these can be fixed in all cases:

```js
foo = ++bar;
baz = --qux;

// is equivalent to:

foo = bar += 1;
baz = qux -= 1;
```